.github/workflows/on-pullrequest-powershell.yml

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: On Pull Request - PowerShell
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**/*.ps1'
+      - '**/*.psm1'
+      - '**/analyze-repository.yml'
+      - '**/powershell-analyzer.yml'
+  workflow_dispatch:
```

Comment:
I did not include a trigger on `push` because directly pushing to the `main` branch is not allowed. Why run the tests again if they've already been successful on the pull request?